packages/ember-views/lib/components/component.js

```diff
@@ -456,6 +456,182 @@ var Component = View.extend(TargetActionSupport, {
     @public
     @property positionalParams
   */
+
+  /**
+    `didInitAttrs` runs on the first render after a component was
```

Comment:
This runs when the component is initialized (aka created), but before render.  Technically, this is called from the components constructor so any properties set here properly shape the object.